.github/workflows/ci.yaml

```diff
@@ -102,10 +102,16 @@ jobs:
       - name: Install Rust
         uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
-          components: llvm-tools-preview
+          cache-workspaces: rust -> target
+
+      - name: Install llvm-tools-preview
+        working-directory: ./rust
+        run: rustup component add llvm-tools-preview
```

Comment:
What is the reason `components: llvm-tools-preview` is not working?

Reply:
It could be because of the `rust/rust-toolchain.toml` file. When this file is there, the component gets installed into `~/.rustup/toolchains/1.80-aarch64-unknown-linux-gnu` instead of the regular `~/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/` (path on a fedora vm on my laptop).
The rust setup action is not doing this correctly since its getting executed from root directory, and we have Rust files inside sub-directory `rust/`.